src/pages/en/core-concepts/astro-components.md

```diff
@@ -143,6 +143,37 @@ const name = "Astro";
 <MyComponent templateLiteralNameAttribute={`MyNameIs${name}`} />
 ```
 
+:::caution
+HTML attributes will be converted to strings, so functions and objects don't work when passed to HTML elements.
+For example, you can't assign an event handler to an HTML element in an Astro component:
+
+```astro
+---
+// dont-do-this.astro
+function handleClick () {
+    console.log("button clicked!");
+}
+---
+<!-- ❌ This doesn't work! ❌ -->
+<button onClick={handleClick}>Nothing will happen when you click me!</button>
+```
+
+Instead, use a client-side script to add the event handler:
```

Comment:
Either here, or in the earlier text, is it worth a tiny call out to "think about how you'd do this in HTML/JS"? This is a great example, both of what people will try to do and what they should do instead. 

The two approaches seem different enough from each other -- one is very "framework-y" where your component is going to render your button and the other is like, just make an html button! One passes props, and the other does not at all. I do wonder if a call out re: "thinking differently" might be helpful.

Reply:
We only want them to think differently in _this_ case (passing an event handler), as a workaround. Other dynamic props still work, so I don't want to overstate the case.

Would you prefer something like this, to invoke the similarity?

> Instead, use a client-side script to add the event handler, like you would in vanilla JavaScript:

Reply:
Yup, I think something like that is perfect!